src/main/java/com/connexta/poller/api/PollingTask.java

```diff
@@ -0,0 +1,17 @@
+/*
+ * Copyright (c) 2019 Connexta, LLC
+ *
+ * Released under the GNU Lesser General Public License version 3; see
+ * https://www.gnu.org/licenses/lgpl-3.0.html
+ */
+package com.connexta.poller.api;
+
+import com.dyngr.core.AttemptMaker;
+import com.dyngr.core.AttemptResult;
+
+/** This interface is the boundary between the polling library and client code. */
+public interface PollingTask<T> extends AttemptMaker<T> {
```

Comment:
❓ Why extend this interface rather than just used the AttemptMaker directly? These both define the same method. We are already coupling our polling API to the library by returning the AttemptResult below.
I'd argue that we should not be exposing a third party library in our API.

Reply:
I went back and forth on it. On the one hand, adding this interface and an impl adds more code and a layer of indirection (added complexity). On the other hand, the PollingTask interface and impl completely insulate other Ion objects from the polling library. AttemptResult can AttemptMaker end up being 100% internal to the PollingTask.  I contemplated which approach was more Java-like. I decided that adding the interface was more Java-like. That said, Java's idioms have never made a lot of sense to me. Things that are best practice in Java are not necessarily best practices in other contexts. If the team feels that another approach is more inline with Java best practices, I''ll gladly change it.

Reply:
Which other Ion objects are interacting with this polling? I figured it would just be a thread that kicks off on system startup and does it's thing for the duration that the process is alive.

Reply:
I expect there will be a single consumer of the polling service-- the one that polls the transform status endpoint. 

Creating interfaces has been drilled into me time so many times that I can no longer use a coffee machine unless it implements the `HotBeverageFactory<T extends HotBeverage>` interface. :coffee: